google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -144,10 +144,15 @@ def list_tables dataset_id, max: nil, token: nil
 
         ##
         # Gets the specified table resource by full table reference.
-        def get_project_table project_id, dataset_id, table_id
+        def get_project_table project_id, dataset_id, table_id, metadata_view: nil
+          metadata_view = table_metadata_view_type_for metadata_view
           # The get operation is considered idempotent
           execute backoff: true do
-            service.get_table project_id, dataset_id, table_id
+            if metadata_view.nil?
+              service.get_table project_id, dataset_id, table_id
+            else
+              service.get_table project_id, dataset_id, table_id, view: metadata_view
```

Comment:
Hi @dazuma, Can you suggest any shorthand for the above if condition? Without this, the existing unit test are failing and updating them manually will take a long time.

Reply:
The `get_table`'s method signature is defined as `def get_table dataset_id, table_id, metadata_view: nil`. So I think I argument to be passed here should be `metadata_view` instead of `view`. I think the overall condition block can be replaced with:
```rb
service.get_table project_id, dataset_id, table_id, metadata_view: metadata_view
```

Reply:
The `service.get_table` method invoked inside `get_project_table` is present the auto-generated layer and not in handwritten layer's `service` class

Reply:
The `service.get_table` method is defined here: https://github.com/googleapis/google-api-ruby-client/blob/main/generated/google-apis-bigquery_v2/lib/google/apis/bigquery_v2/service.rb#L1412

Its definition is:

```ruby
def get_table(project_id, dataset_id, table_id, selected_fields: nil, view: nil, fields: nil, quota_user: nil, user_ip: nil, options: nil, &block)
```

So you should just be able to pass `metadata_view` directly to the `view` parameter:

```ruby
execute backoff: true do
  service.get_table project_id, dataset_id, table_id, view: metadata_view
end
```

Reply:
Yes, but the current unit tests will fail with this approach because they dont expect the view parameter. So the choice was to either update the existing unit test or add an if block.

Reply:
code refactored in #19911 to solve this issue